Hakawai/Mentions/HKWMentionsStartDetectionStateMachine.m

```diff
@@ -253,21 +258,14 @@ - (void)deleteTypedCharacter:(unichar)deletedChar withCharacterNowPrecedingCurso
 
 - (void)cursorMovedWithCharacterNowPrecedingCursor:(unichar)c {
     // Determine the type of the character
-    enum CharacterType {
-        CharacterTypeSeparator = 0,
-        CharacterTypeNormal
-    };
-    enum CharacterType currentCharacterType = CharacterTypeNormal;
-    if ([HKWMentionsStartDetectionStateMachine.separatorSet characterIsMember:c] || c == 0) {
-        currentCharacterType = CharacterTypeSeparator;
-    }
+    enum CharacterType currentCharacterType = [self characterTypeOfCharacter:c];
     switch (self.state) {
         case HKWMentionsStartDetectionStateQuiescentReady:
         case HKWMentionsStartDetectionStateQuiescentStalled: {
             // Reset the string buffer
             self.stringBuffer = [@"" mutableCopy];
             self.charactersSinceLastWhitespace = 0;
-            if (currentCharacterType == CharacterTypeSeparator) {
+            if (currentCharacterType == CharacterTypeSeparator || currentCharacterType == CharacterTypeControlCharacter) {
                 // The user moved the cursor to the beginning of the text region, or right after a newline or whitespace or punctuation
```

Comment:
update

Reply:
From what we could understand of the comment, this condition to set state as quiescent ready is the current character has to be a punctuation character or whitespace. For the case of '@', it is a control character as well as a punctuation character and `characterTypeOfCharacter` returns control character type and not separator.
To cover both control characters and punctuations, we have added such check. 
Please let us know if we misunderstood the comment.

Reply:
no i just meant update this comment: "right after a newline or whitespace or punctuation" to include talking about cntrl character